tekton/pipeline_run.go

```diff
@@ -123,9 +124,10 @@ func (r *ReleasePipelineRun) WithObjectReferences(objects ...client.Object) *Rel
 	return r
 }
 
-// WithOwner set's owner annotations to the release PipelineRun.
+// WithOwner set's owner annotations to the release PipelineRun and a finalizer to prevent its deletion.
```

Comment:
Nit: `s/set's/sets/`

Also, wouldn't it be nicer to have a separate `WithFinalizer` function for this?

Reply:
Agree with `s/set's/sets/`. I think keeping it all in one function is okay though. I kind of think the finalizer concept fits under the English meaning of ownership. Fine with it being split up too though